File: app/src/main/java/com/example/installer/api/ApiService.java
package com.example.installer.api;


import com.example.installer.entity.APKEntity;
import com.example.installer.entity.PackageEntity;
import com.example.installer.entity.Result;

import java.util.List;

import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.Query;
import rx.Observable;

/**
 * Author:  ZhangTao
 * Date: 2018/3/5.
 */

public interface ApiService {
    /*获取所有APP列表*/
    @Headers("Cache-Control:0")
    @GET("/applicationClient/getApplications")
    Observable<Result<List<PackageEntity>>> getApplicationList();

    /*获取指定平台的特定APK(release/debug)列表*/
    @Headers("Cache-Control:0")
    @GET("/applicationClient/getApplicationVersions")
    Observable<Result<List<APKEntity>>> getPackageList(@Query("system_name") String system_name, @Query("application_id") String application_id, @Query("version_type") String version_type, @Query("page") int pageIndex);

}

File: app/src/main/java/com/example/installer/api/APIService.kt
package com.example.installer.api

import com.example.installer.entity.PackageEntity
import com.example.installer.entity.Result
import com.example.installer.entity.ProductEntity
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.Query
import rx.Observable

/**
@author zhangtao
@date   2020/11/2

 **/
interface APIService {
    /*获取所有APP列表*/
    @Headers("Cache-Control:0")
    @GET("/applicationClient/getApplications")
    fun getProductList(): Observable<Result<List<ProductEntity>>>

    /*获取指定平台的特定APK(release/debug)列表*/
    @Headers("Cache-Control:0")
    @GET("/applicationClient/getApplicationVersions")
    fun getPackageList(
        @Query("system_name") system_name: String?,
        @Query("application_id") application_id: String?,
        @Query("version_type") version_type: String?,
        @Query("page") pageIndex: Int
    ): Observable<Result<List<PackageEntity>>>

}